core/src/main/java/org/apache/iceberg/BaseFile.java

```diff
@@ -60,7 +60,7 @@ public PartitionData copy() {
   private String filePath = null;
   private FileFormat format = null;
   private PartitionData partitionData = null;
-  private Long recordCount = null;
+  private long recordCount = -1L;
```

Comment:
I don't think we should make this change. I like that the `NullPointerException` ensures that an incorrect record count just can't be used.

Reply:
I mostly did this to simplify testing as I noticed `fileSizeInBytes` which is also a `long` has the pattern of defaulting to -1, but I don't have strong opinion either way. Will revert!